tests/generic_config_updater/test_dynamic_acl.py

```diff
@@ -0,0 +1,755 @@
+
+
+def dynamic_acl_create_three_drop_rules(duthost, setup):
+    """Create 3 drop rules in the format required when an ACL table does not have any rules in it yet"""
+
+    extra_vars = {
+        'blocked_port_1': setup["scale_port_names"][0],
+        'blocked_port_2': setup["scale_port_names"][1],
+        'blocked_port_3': setup["scale_port_names"][2]
+
+    }
+
+    output = format_and_apply_template(duthost, CREATE_THREE_DROP_RULES_TEMPLATE, extra_vars)
+
+    expected_rule_3_content = ["DYNAMIC_ACL_TABLE",
+                               "RULE_3",
+                               "9997",
+                               "DROP",
+                               "IN_PORTS: " + extra_vars['blocked_port_1'],
+                               "Active"]
+    expected_rule_4_content = ["DYNAMIC_ACL_TABLE",
+                               "RULE_4",
+                               "9996",
+                               "DROP",
+                               "IN_PORTS: " + extra_vars['blocked_port_2'],
+                               "Active"]
+    expected_rule_5_content = ["DYNAMIC_ACL_TABLE",
+                               "RULE_5",
+                               "9995",
+                               "DROP",
+                               "IN_PORTS: " + extra_vars['blocked_port_3'],
+                               "Active"]
+
+    expect_op_success(duthost, output)
+
+    expect_acl_rule_match(duthost, "RULE_3", expected_rule_3_content)
+    expect_acl_rule_match(duthost, "RULE_4", expected_rule_4_content)
+    expect_acl_rule_match(duthost, "RULE_5", expected_rule_5_content)
+
+
+def dynamic_acl_verify_packets(setup, ptfadapter, packets, packets_dropped, src_port=None):
+    """Verify that the given packets are either dropped/forwarded correctly
+
+    Args:
+        packets: the packets that we are sending
+        packets_dropped: whether or not we are expecting to drop or forward these packets
+        src_port_blocked: whether or not to send it on the source port that we block in our drop rules"""
+    if packets_dropped:
+        action_type = "dropped"
+    else:
+        action_type = "forwarded"
+
+    if src_port is None:
+        src_port = setup["blocked_src_port_indice"]
+
+    for rule, pkt in list(packets.items()):
+        logger.info("Testing that {} packets are correctly {}".format(rule, action_type))
+        exp_pkt = build_exp_pkt(pkt)
+        # Send and verify packet
+        ptfadapter.dataplane.flush()
+        testutils.send(ptfadapter, pkt=pkt, port_id=src_port)
+        verify_expected_packet_behavior(exp_pkt, ptfadapter, setup, expect_drop=packets_dropped)
+
+
+def dynamic_acl_remove_third_drop_rule(duthost):
+    """Remove the third drop rule of the three created for the drop rule removal test"""
+    with open(os.path.join(TEMPLATES_DIR, REMOVE_THIRD_DROP_RULE_FILE)) as file:
+        json_patch = json.load(file)
+
+    tmpfile = generate_tmpfile(duthost)
+    logger.info("tmpfile {}".format(tmpfile))
```

Comment:
Minor suggestion, the code snippet is used in multiple functions, it should be better if wrap into a helper function.

Reply:
moved to reusable function